backend/src/main/java/com/google/sps/servlets/AddParticipantServlet.java

```diff
@@ -99,27 +92,34 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       return;
     }
 
-    // id is irrelevant, only relevant when getting from datastore
+    // Create new Participant from input parameters
+    // id is irrelevant, created when added to datastore
     Participant newParticipant =
         new Participant(
-            /* id= */ -1L, username, startTimeAvailable, endTimeAvailable, duration, timestamp);
+            username, startTimeAvailable, endTimeAvailable, duration, /* matchId=*/ 0, timestamp);
 
+    // Get DatastoreService and instiate Match and Participant Datastores
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    MatchDatastore matchDatastore = new MatchDatastore(datastore);
+    ParticipantDatastore participantDatastore = new ParticipantDatastore(datastore);
 
     // Find immediate match if possible
-    FindMatchQuery query = new FindMatchQuery(Clock.systemUTC());
-    Match match = query.findMatch(getParticipants(datastore), newParticipant);
+    FindMatchQuery query = new FindMatchQuery(Clock.systemUTC(), participantDatastore);
+    Match match = query.findMatch(newParticipant);
 
-    // Match found, add to datastore, delete matched participants from datastore
     if (match != null) {
-      addMatchToDatastore(match, datastore);
-      deleteParticipantFromDatastore(match.getSecondParticipant(), datastore);
+      // Match found, add to match datastore, update participant datastore
+      long matchId = matchDatastore.addMatch(match);
+
+      // Update participant entities with new matchId and null availability
+      Participant currParticipant =
+          participantDatastore.getParticipantFromUsername(match.getSecondParticipantUsername());
+      nullAvailabilityAdd(participantDatastore, currParticipant, matchId);
+      nullAvailabilityAdd(participantDatastore, newParticipant, matchId);
     } else {
-      datastore.put(addParticipantEntityToDatastore(newParticipant));
+      // Match not found, add participant to datastore
+      participantDatastore.addParticipant(newParticipant);
     }
-
-    response.setContentType("text/plain;charset=UTF-8");
-    response.getWriter().println("Received form input details!");
```

Comment:
WDYT about leaving these lines in the servlet for testing purposes just as a confirmation to the frontend that the form details were recieved?

Reply:
Oops, didn't mean to delete that! Thanks for catching that!